src/helper/subscriber/base.hpp

```diff
@@ -42,6 +42,9 @@ class subscriber {
 
     virtual std::vector<std::string_view> get_subscriptions() = 0;
     virtual listener::ptr get_listener() = 0;
+
+protected:
+    subscriber() = default;
```

Comment:
Since the class can't be instantiated, what was the purpose of making this protected?

Reply:
Right, doesn't make much of a difference because it's an abstract type (could be public). It could be argued it better expresses the idea it's called only by subclasses, but I can leave it out.

Reply:
Right, it being public has the same effect. It could be argued that it better expresses the idea that it's only called by subclasses, but I'll leave it out for some economy of code.